### ISSUE TITLE
Type-mask: contract violation expected: Type? given: #f

### ISSUE THREAD

**Issue by jsmaniac** [racket/typed-racket#447]
### What version of Racket are you using?
6.7.0.1 (20161022-153dc01)

### What program did you run?
```
#lang typed/racket
(struct St ())
(define-type A (∀ (B) (→ (∩ B St) Number)))
```

### What should have happened?

The program should compile (it only defines a struct and what looks like a valid type to me).

### If you got an error message, please include it here.

```
.racket/snapshot/pkgs/typed-racket-lib/typed-racket/infer/intersect.rkt:24:2: Type-mask: contract violation
  expected: Type?
  given: #f
```

**Comment by jsmaniac** [comment]
FWIW, this equivalent type compiles fine:

```
#lang typed/racket
(struct St () #:transparent)
(define-type A ((∀ (S) (∀ (B) (→ (∩ B S) Number))) St))
```

**Comment by pnwamk** [comment]
fixed by 9df3a1d
